core/runtime/src/main/java/io/quarkus/runtime/Quarkus.java

```diff
@@ -182,6 +183,12 @@ public static void waitForExit() {
      * Must not be called by the main thread, or a deadlock will result.
      */
     public static void blockingExit() {
+        if (Thread.currentThread().getThreadGroup().getName().equals("main") &&
```

Comment:
I think this will be ok. If we get reports of false positives though we should probably move it to some kind of ThreadLocal marker.

Reply:
Definitely